web/modules/custom/dbcdk_community/src/Plugin/Block/ProfilesBlock.php

```diff
@@ -0,0 +1,277 @@
+ * )
+ */
+class ProfilesBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  use CommunityTraits;
+
+  /**
+   * The current request stack.
+   *
+   * @var RequestStack $requestStack
+   */
+  protected $requestStack;
+
+  /**
+   * The DBCDK Community Service Profile API.
+   *
+   * @var ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The amount of items to be shown on each page.
+   *
+   * @var int $pagerLimit
+   */
+  protected $pagerLimit;
+
+  /**
+   * Creates a Profiles Block instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param RequestStack $request_stack
+   *   The current request.
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The DBCDK Community Service Profile API.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, RequestStack $request_stack, ProfileApi $profile_api) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->requestStack = $request_stack;
+    $this->profileApi = $profile_api;
+    $this->pagerLimit = (!empty($this->configuration['pager_limit']) ? $this->configuration['pager_limit'] : 25);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('request_stack'),
+      $container->get('dbcdk_community.api.profile')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form['pager_limit'] = array(
+      '#type' => 'number',
+      '#title' => $this->t('Pager limit'),
+      '#description' => $this->t('The amount of items to be shown on each page.'),
+      '#maxlength' => 255,
+      '#default_value' => $this->configuration['pager_limit'],
+    );
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    if (!$form_state->getErrors()) {
+      $this->configuration['pager_limit'] = $form_state->getValue('pager_limit');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // Tries to fetch profiles from the Community Service.
+    // We catch any API Exceptions and stop the rest of the code from executing
+    // so we don't get fatal errors the will cause a "status code 500" but
+    // rather displays an empty table.
+    try {
+      $filter = [
+        'limit' => $this->pagerLimit,
+        'offset' => $this->requestStack->getCurrentRequest()->query->get('page') * $this->pagerLimit,
```

Comment:
If the `pager_default_initialize()` function gets refactored to "not being a global thingy" and we have to refactor the code later on (thinking Drupal 8.2.x, 8.3.x...), then there would be less work in changing it because we got the page-value the "correct" way to begin with?